server/logout/logout.go

```diff
@@ -60,18 +61,33 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	logoutRedirectURL := strings.TrimRight(strings.TrimLeft(argoCDSettings.URL, "/"), "/") + strings.TrimRight(strings.TrimLeft(h.rootPath, "/"), "/")
 
-	argocdCookie, err := r.Cookie(common.AuthCookieName)
-	if err != nil {
+	cookies := r.Cookies()
+	// filter cookies with prefix common.AuthCookieName
+	var filteredCookies []string
+	for _, cookie := range cookies {
+		if strings.HasPrefix(cookie.Name, common.AuthCookieName) {
+			filteredCookies = append(filteredCookies, fmt.Sprintf("%s=%s", cookie.Name, cookie.Value))
+		}
+	}
+	tokenString, err = httputil.JoinCookies(common.AuthCookieName, strings.Join(filteredCookies, "; "))
```

Comment:
Looks like we join cookie into one string and `JoinCookies` immediately split it. Why not just pass list of strings?

Reply:
We can pass the list of strings as well. I kept this format as in server.go we get token in this format from `metadata.MD` https://github.com/argoproj/argo-cd/pull/5497/files?file-filters%5B%5D=.go#diff-78f42ba40d0f10b08c73b7e6bb8376f398e249c963cf549e89591d6b6826b9a4R938